lib/manageiq/environment.rb

```diff
@@ -54,9 +53,13 @@ def self.while_updating_ui
       end
 
       ui_thread.abort_on_exception = true
+      ui_thread
+    end
 
+    def self.while_updating_ui
+      # Run update:ui in a thread and continue to do the non-js stuff
+      ui_thread = update_ui_thread
```

Comment:
Wondering if the CI check should just live inside this method and then leave the block-based usage above the same.

Reply:
Yeah the only reason I didn't do that was because `ManageIQ::Environment` isn't only used by core, and something else might want to use `while_updating_ui` and actually expect it to update the UI :)

Reply:
Makes sense.  We can drop the method after this is merged if no one else is calling it.